app/Http/Controllers/WelcomeController.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Cache;
+use DB;
+
+class WelcomeController extends Controller
+{
+    private $_dictionary;
+    
+    /**
+     * @param  Request $request
+     * @return Factory|View
+     */
+    public function index()//(Request $request)
+    {
+        return view(
+            'welcome',
+            [
+                'subjects' => $this->_dictionary,
+            ]
+        );
+    }
+
+    /**
+     * @param Request $request
+     */
+    public function cache(Request $request)
+    {
+        if ($cacheTime = Cache::get('profileCacheTime')) {
+            return response()->json(
+                [
+                'time' => $cacheTime,
+                ]
+            );
+        }
+
+        $now = Carbon::now();
+        Cache::set('profileCacheTime', $now, 60);
+
+        return response()->json(
+            [
+            'time' => $now,
+            ]
+        );
+    }
+
+    public function __construct()
+    {
+        $data = DB::select('select * from course');
+
+        foreach ($data as $elem) {
+            $arr = [];
+            $index = 0;
+            foreach ($elem as $key => $value) {
+                $arr[ $key ] = $value;
+                $index += 1;
+
+                if ($index >= 4) {
+                    break;
+                }
+            }
+
+            $this->_dictionary[] = $arr;   
+        }
```

Comment:
* 請直接使用建立的 Model 處理
* 這邊的邏輯取前幾個欄位，如果順序發生變化，可能會有錯誤，這邊建議限定取得的欄位
```suggestion
     $this->_dictionary = Course::get(['id', 'name'])->toArray();
```

Reply:
- 已應用並修正內容